### PR TITLE
Vendor gcplogs deps

### LINKED ISSUE #25403: update Google Cloud API client import paths and more
The Google Cloud API client libraries for Go are making some breaking changes:
- The import paths are changing from `google.golang.org/cloud/...` to
  `cloud.google.com/go/...`. For example, if your code imports the BigQuery client
  it currently reads
  `import "google.golang.org/cloud/bigquery"`
  It should be changed to
  `import "cloud.google.com/go/bigquery"`
- Client options are also moving, from `google.golang.org/cloud` to
  `google.golang.org/api/option`. Two have also been renamed:
  - `WithBaseGRPC` is now `WithGRPCConn`
  - `WithBaseHTTP` is now `WithHTTPClient`
- The `cloud.WithContext` and `cloud.NewContext` methods are gone, as are the
  deprecated pubsub and container functions that required them. Use the `Client`
  methods of these packages instead.

You should make these changes before September 12, 2016, when the packages at
`google.golang.org/cloud` will go away.

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**

Updated gcplogs deps.

The aim of this PR is to vendor https://github.com/golang/oauth2/commit/96382aa079b72d8c014eb0c50f6c223d1e6a2de0 for PR https://github.com/docker/docker/pull/29478, so as to fix issue https://github.com/docker/docker/issues/29344.

Also closes https://github.com/docker/docker/issues/25403

**- How I did it**

The first commit vendors the latest oauth2 and its deps.
So basically we only need the first commit.

However, the first commit is not enough because it vendors both `google.golang.org/cloud` and `cloud.google.com/go`.
So the second commit removes `google.golang.org/cloud` and completely migrates to `cloud.google.com/go`.

**- How to verify it**

`docker run --log-driver=gcplogs ...`

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
cc @mikedanese

**Comment by thaJeztah** [comment]
ping @justincormack as well 👍

File: vendor.conf [hunk 1]
```diff
@@ -88,9 +88,12 @@ github.com/jmespath/go-jmespath 0b12d6b521d83fc7f755e7cfc1b1fbdd35a01a74
 github.com/bsphere/le_go d3308aafe090956bc89a65f0769f58251a1b4f03
 
 # gcplogs deps
-golang.org/x/oauth2 2baa8a1b9338cf13d9eeb27696d761155fa480be
+golang.org/x/oauth2 96382aa079b72d8c014eb0c50f6c223d1e6a2de0
 google.golang.org/api dc6d2353af16e2a2b0ff6986af051d473a4ed468
+# TODO: remove google.golang.org/cloud, which has been replaced by cloud.google.com/go
```
Comment (thaJeztah) [inline vendor.conf]: oh, we have an issue for that https://github.com/docker/docker/issues/25403

**Comment by thaJeztah** [comment]
Looks like https://github.com/docker/docker/issues/25403 will be fixed by this as well, or are there more instances?

**Comment by AkihiroSuda** [comment]
@thaJeztah Yes, this PR will close https://github.com/docker/docker/issues/25403 as well

**Review by justincormack — APPROVED** [review]

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**